athenz/assertion.go

```diff
@@ -109,19 +125,58 @@ func expandPolicyAssertions(dn string, configured []interface{}) []*zms.Assertio
 
 func flattenPolicyAssertion(list []*zms.Assertion) []interface{} {
 	policyAssertions := make([]interface{}, 0, len(list))
+
 	for _, a := range list {
 		role := strings.Split(a.Role, ROLE_SEPARATOR)[1]
 		resource := a.Resource
 		effect := a.Effect.String()
 		action := a.Action
+		caseSensitive := inferCaseSensitiveValue(action, resource)
 
 		a := map[string]interface{}{
-			"role":     role,
-			"resource": resource,
-			"action":   action,
-			"effect":   effect,
+			"role":           role,
+			"resource":       resource,
+			"action":         action,
+			"effect":         effect,
+			"case_sensitive": caseSensitive,
 		}
 		policyAssertions = append(policyAssertions, a)
+
 	}
+
 	return policyAssertions
 }
+
+// we're not going to allow to enabled case_sensitive within an assertion if didn't specify any upper case characters in action or resource
+func validateCaseSensitiveValue(caseSensitive bool, action string, resourceName string) error {
+	if caseSensitive {
+		if strings.ToLower(resourceName) == resourceName && action == strings.ToLower(action) {
+			return fmt.Errorf("you cannot enbled case_sensitive flag if you didn't specify any upper case characters in action or resource")
+		}
+	}
+	return nil
+}
+
+// to infer from assertion action and assertion resource the value of case_sensitive
+func inferCaseSensitiveValue(action, resourceName string) bool {
```

Comment:
inferring the case_sensitive flag is relevant only for data source.
if we are going to create or update existing resource we should know its definition.